### PR TITLE
Emotion subsystem

### PR DESCRIPTION
**Pull request by dcam0050**
Addition of an emotion subsystem to control iCub face emotions from within icub-client. Also changed return of SAM classification from tuple to yarp for python compatibility.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Cool, thanks. I didn't realize we haven't ported the emotion subsystem from wysiwyd to icub-hri. I'll review this next week after the CVPR deadline.

Best, Tobi

File: src/libraries/icubclient/include/icubclient/subsystems/subSystem_emotion.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * @return true if command was sent successful.
+     */
+    bool setEmotion(Emotion emotion, Part part);
+
+    /**
+     * @brief Method to switch iCub emotions for whole face.
+     * @param emotion Selection from Emotion enum
+     * @return true if command was sent successful.
+     */
+    bool setEmotion(Emotion emotion);
```
Comment (Tobias-Fischer) [inline src/libraries/icubclient/include/icubclient/subsystems/subSystem_emotion.h]: Can you add a `getEmotion` method?

File: src/libraries/icubclient/src/subsystems/subSystem_emotion.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+bool icubclient::SubSystem_emotion::setEmotion(Emotion emotion) {
+    return setEmotion(emotion, Part::all);
+}
+
+bool icubclient::SubSystem_emotion::setEmotion(Emotion emotion, Part part) {
+    yarp::os::Bottle bReq, bResp;
+    std::string spart, semotion;
+
+    switch(emotion)
```
Comment (Tobias-Fischer) [inline src/libraries/icubclient/src/subsystems/subSystem_emotion.cpp]: This is really ugly. Why don't you create a `std::pair` which contains the pairs?

**Comment by dcam0050** [comment]
Thanks for the feedback. However I don't understand what you mean by using std::pair.

**Comment by dcam0050** [comment]
I haven't tested correct operation yet but is this what you meant?

**Comment by Tobias-Fischer** [comment]
Hey,
Yes this is kind of what I meant :-). I just committed a change to hopefully get this branch compiled under Windows as well. Have you tested operation on the robot yet? As soon as it compiles on Windows and is tested on the robot, I am happy to merge.

Best, Tobi

**Comment by dcam0050** [comment]
I tested it before my previous commit :) I'll pull with you changes and double check but should be ready to go

**Comment by Tobias-Fischer** [comment]
Cool. Just let me know once it's tested and I'll merge.

**Comment by dcam0050** [comment]
Tested and works!

File: src/libraries/icubclient/include/icubclient/subsystems/subSystem_SAM.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ class SubSystem_SAM : public SubSystem
      * @param model Model name to trigger a classification for
      * @return tuple of <bool, std::string> with success as bool and classification as std::string
```
Comment (Tobias-Fischer) [inline src/libraries/icubclient/include/icubclient/subsystems/subSystem_SAM.h]: Please update the docstring for this method.

**Comment by Tobias-Fischer** [comment]
Thanks a lot!

**Comment by dcam0050** [comment]
:+1:

**Pull request merged**